src/data/mv_polynomial.lean

```diff
@@ -1298,4 +1329,115 @@ end
 
 end equiv
 
+/-!
+## Partial derivatives
+-/
+section pderivative
+
+variables {R : Type} [comm_ring R]
+variable {S : Type}
+
+/-- `pderivative v p` is the partial derivative of `p` with respect to `v` -/
+def pderivative (v : S) (p : mv_polynomial S R) : mv_polynomial S R :=
+p.sum (λ A B, monomial (A - single v 1) (B * (A v)))
+
+
+lemma pderivative_add {v : S} {f g : mv_polynomial S R} :
+  pderivative v (f + g) = pderivative v f + pderivative v g :=
+begin
+  refine sum_add_index _ _,
+  { intro, simpa [monomial] },
+  intros, rw [monomial_add, add_mul],
+end
```

Comment:
* No need for `intro` in either branch.
* Second half of proof needs braces, too, for readability.
* `simpa [monomial]` unnecessary, you're not actually using an argument. `simp` suffices.
* `rw [monomial_add, add_mul]` could just be `simp [add_mul]`, since `monomial_add` is already a `simp` lemma.
* This lemma should be marked as `@[simp]`.